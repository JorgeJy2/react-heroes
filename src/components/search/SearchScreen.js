import React, { useMemo, useState } from 'react'
import { useLocation } from 'react-router';
import { getHeroesByName } from '../../selectors/getHeroeByName';
import { HeroeCard } from '../heroes/HeroeCard';
import queryString from 'query-string';


export const SearchScreen = ({ history }) => {

    const location = useLocation();
    const { q = '' } = queryString.parse(location.search);

    const [searchHeroe, setsearchHeroe] = useState('');

    const heroesFiltered = useMemo(() => getHeroesByName(q), [q]);


    const handleForm = (e) => {
        e.preventDefault();

        history.push(`?q=${searchHeroe}`);
    };

    return (
        <div>
            <h1>Search</h1>
            <hr />
            <div className="row">
                <div className="col-5">
                    <h4>Search form</h4>
                    <hr />
                    <form onSubmit={handleForm}>
                        <input
                            type="text"
                            placeholder="Find your hero"
                            className="form-control"
                            value={searchHeroe}
                            onChange={(event) => setsearchHeroe(event.target.value)}
                        />
                        <button
                            type="submit"
                            className="btn mt-2 btn-block btn-outline-primary"

                        >Search</button>
                    </form>
                </div>
                <div className="col-7">
                    <h4>Result</h4>

                    {
                        (heroesFiltered.length === 0) 
                            && 
                            <p> Heroe not found</p>
                    }

                    {
                        heroesFiltered.map(heroe => (
                            <HeroeCard
                                key={heroe.id}
                                className="animate__animated animate__fadeInLeft"
                                {...heroe} />
                        ))
                    }
                </div>
            </div>
        </div>
    )
}
