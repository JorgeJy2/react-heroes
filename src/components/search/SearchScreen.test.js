import { mount} from "enzyme";
import { MemoryRouter, Route } from "react-router";
import { SearchScreen } from "./SearchScreen";

describe('test in search', ()=>{
    test('should snaptshot', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search']}>
                <Route  path="/search" component={SearchScreen}  />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Search');
    });


    test('should find batman ', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <Route  path="/search" component={SearchScreen}  />
            </MemoryRouter>
        );
        expect(wrapper.find('input').prop('value')).toBe('batman');
        expect(wrapper.find('.card').length).toBe(1);
    });


    test('should not found', () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=not-found']}>
                <Route  path="/search" component={SearchScreen}  />
            </MemoryRouter>
        );
        expect(wrapper.find('input').prop('value')).toBe('not-found');
        expect(wrapper.find('.card').length).toBe(0);
        expect(wrapper.find('p').text().trim()).toBe('Heroe not found');
    });

    test('should submit', () => {

        const history ={
            push: jest.fn()
        }

        const wrapper = mount(
            <MemoryRouter initialEntries={['/search?q=not-found']}>
                <Route path="/search"
                        component={ ()=>
                            <SearchScreen history={history} />}
                              />
            </MemoryRouter>
        );

        wrapper.find('input').simulate('change', {
            target: {
                value: 'batman'
            }
        });

        const formSubmit = wrapper.find('form').prop('onSubmit');
        formSubmit({ preventDefault() { } });

        expect(history.push).toHaveBeenCalledWith('?q=batman');
    });
    
    
});