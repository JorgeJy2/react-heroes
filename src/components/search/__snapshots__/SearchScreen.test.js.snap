// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in search should snaptshot 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Find your hero"
          type="text"
          value=""
        />
        <button
          className="btn mt-2 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Result
      </h4>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Batman"
              className="card-img"
              src="./assets/heroes/dc-batman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Batman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Wayne
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #27
                </small>
              </p>
              <a
                href="/heroe/dc-batman"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Superman"
              className="card-img"
              src="./assets/heroes/dc-superman.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Superman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Kal-El
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Action Comics #1
                </small>
              </p>
              <a
                href="/heroe/dc-superman"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Flash"
              className="card-img"
              src="./assets/heroes/dc-flash.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Flash
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Jay Garrick
                 
              </p>
              <p
                className="card-text"
              >
                Jay Garrick, Barry Allen, Wally West, Bart Allen
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Flash Comics #1
                </small>
              </p>
              <a
                href="/heroe/dc-flash"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Green Lantern"
              className="card-img"
              src="./assets/heroes/dc-green.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Green Lantern
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Alan Scott
                 
              </p>
              <p
                className="card-text"
              >
                Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  All-American Comics #16
                </small>
              </p>
              <a
                href="/heroe/dc-green"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Green Arrow"
              className="card-img"
              src="./assets/heroes/dc-arrow.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Green Arrow
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Oliver Queen
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  More Fun Comics #73
                </small>
              </p>
              <a
                href="/heroe/dc-arrow"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Wonder Woman"
              className="card-img"
              src="./assets/heroes/dc-wonder.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Wonder Woman
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Princess Diana
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  All Star Comics #8
                </small>
              </p>
              <a
                href="/heroe/dc-wonder"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Martian Manhunter"
              className="card-img"
              src="./assets/heroes/dc-martian.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Martian Manhunter
                 
              </h5>
              <p
                className="card-text"
              >
                 
                J'onn J'onzz
                 
              </p>
              <p
                className="card-text"
              >
                Martian Manhunter
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #225
                </small>
              </p>
              <a
                href="/heroe/dc-martian"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Robin/Nightwing"
              className="card-img"
              src="./assets/heroes/dc-robin.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Robin/Nightwing
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Dick Grayson
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Detective Comics #38
                </small>
              </p>
              <a
                href="/heroe/dc-robin"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Blue Beetle"
              className="card-img"
              src="./assets/heroes/dc-blue.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Blue Beetle
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Dan Garret
                 
              </p>
              <p
                className="card-text"
              >
                Dan Garret, Ted Kord, Jaime Reyes
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Mystery Men Comics #1
                </small>
              </p>
              <a
                href="/heroe/dc-blue"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Black Canary"
              className="card-img"
              src="./assets/heroes/dc-black.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Black Canary
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Dinah Drake
                 
              </p>
              <p
                className="card-text"
              >
                Dinah Drake, Dinah Lance
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Flash Comics #86
                </small>
              </p>
              <a
                href="/heroe/dc-black"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Spider Man"
              className="card-img"
              src="./assets/heroes/marvel-spider.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Spider Man
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Peter Parker
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Amazing Fantasy #15
                </small>
              </p>
              <a
                href="/heroe/marvel-spider"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Captain America"
              className="card-img"
              src="./assets/heroes/marvel-captain.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Captain America
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Steve Rogers
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Captain America Comics #1
                </small>
              </p>
              <a
                href="/heroe/marvel-captain"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Iron Man"
              className="card-img"
              src="./assets/heroes/marvel-iron.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Iron Man
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Tony Stark
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Tales of Suspense #39
                </small>
              </p>
              <a
                href="/heroe/marvel-iron"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Thor"
              className="card-img"
              src="./assets/heroes/marvel-thor.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Thor
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Thor Odinson
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Journey into Myster #83
                </small>
              </p>
              <a
                href="/heroe/marvel-thor"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Hulk"
              className="card-img"
              src="./assets/heroes/marvel-hulk.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Hulk
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Bruce Banner
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #1
                </small>
              </p>
              <a
                href="/heroe/marvel-hulk"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Wolverine"
              className="card-img"
              src="./assets/heroes/marvel-wolverine.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Wolverine
                 
              </h5>
              <p
                className="card-text"
              >
                 
                James Howlett
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Incredible Hulk #180
                </small>
              </p>
              <a
                href="/heroe/marvel-wolverine"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Daredevil"
              className="card-img"
              src="./assets/heroes/marvel-daredevil.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Daredevil
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Matthew Michael Murdock
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Daredevil #1
                </small>
              </p>
              <a
                href="/heroe/marvel-daredevil"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Hawkeye"
              className="card-img"
              src="./assets/heroes/marvel-hawkeye.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Hawkeye
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Clinton Francis Barton
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  Tales of Suspense #57
                </small>
              </p>
              <a
                href="/heroe/marvel-hawkeye"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Cyclops"
              className="card-img"
              src="./assets/heroes/marvel-cyclops.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Cyclops
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Scott Summers
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  X-Men #1
                </small>
              </p>
              <a
                href="/heroe/marvel-cyclops"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="card ms-3"
        style={
          Object {
            "maxWidth": 540,
          }
        }
      >
        <div
          className="row no-gutters"
        >
          <div
            className="col-md-4"
          >
            <img
              alt="Silver Surfer"
              className="card-img"
              src="./assets/heroes/marvel-silver.jpg"
            />
          </div>
          <div
            className="col-md-8"
          >
            <div
              className="card-body"
            >
              <h5
                className="card-title"
              >
                 
                Silver Surfer
                 
              </h5>
              <p
                className="card-text"
              >
                 
                Norrin Radd
                 
              </p>
              <p
                className="card-text"
              >
                <small
                  className="text-muted"
                >
                  The Fantastic Four #48
                </small>
              </p>
              <a
                href="/heroe/marvel-silver"
                onClick={[Function]}
              >
                More
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
